src/network/taraxa_capability.cpp

```diff
@@ -1511,48 +1510,61 @@ void TaraxaCapability::sendPbftBlocks(NodeID const &_id, size_t height_to_sync,
-    // MAX_PACKET_SIZE (15 MB) limit
-    if (act_packet_size + pbft_block_rlp.out().size() > MAX_PACKET_SIZE) {
-      if (pbft_blocks_rlps.empty()) {
-        // If this log is present in log, it means there was created such big pbft block, that we are not able to send
-        // it through network
-        // This situation should never ever happen !!!
-        LOG(log_er_) << "Unable to send pbft block: " << pbft_blocks[pbft_block_idx].pbft_blk->getBlockHash().abridged()
-                     << " due to MAX_PACKET_SIZE(" << MAX_PACKET_SIZE << ") limit. "
-                     << "Block size: " << pbft_block_rlp.out().size() << " [B]";
-        // Skip this block as it would not go through network anyway
-        continue;
-      }
+      // When checking if size limit exceeds MAX_PACKET_SIZE there are few bytes or rlp structure that is added
+      // for the pbft block and dag block. This should be just a couple of bytes but we enforece even stricter 128 limit
+      static const int RLP_OVERHEAD = 128;
+
+      // Check if PBFT blocks need to be split and sent in multiple packets so we dont exceed
+      // MAX_PACKET_SIZE (15 MB) limit
+      if (act_packet_size + dag_blocks_size + dag_rlp.out().size() + pbft_blocks[pbft_block_idx].rlp().size() +
+              RLP_OVERHEAD >
+          MAX_PACKET_SIZE) {
+        LOG(log_dg_dag_sync_) << "Sending partial PbftBlockPacket due tu MAX_PACKET_SIZE limit. " << pbft_block_idx + 1
+                              << " blocks out of " << pbft_blocks.size() << " sent.";
+
+        pbft_block_rlp.appendList(1);  // Only list of dag blocks and no pbft block rlp for incomplete packet
+        pbft_block_rlp.appendList(dag_blocks_rlps.size());
+
+        for (auto &dag_block : dag_blocks_rlps) pbft_block_rlp.appendRaw(dag_block);
 
-      LOG(log_dg_dag_sync_) << "Sending partial PbftBlockPacket due tu MAX_PACKET_SIZE limit. " << pbft_block_idx + 1
-                            << " blocks out of " << pbft_blocks.size() << " sent.";
+        pbft_blocks_rlps.emplace_back(pbft_block_rlp.invalidate());
 
-      // Send partial packet
-      sealAndSend(_id, PbftBlockPacket, create_packet(std::move(pbft_blocks_rlps)));
+        // Send partial packet
+        sealAndSend(_id, PbftBlockPacket, create_packet(std::move(pbft_blocks_rlps)));
 
-      act_packet_size = 0;
-      pbft_blocks_rlps.clear();
+        act_packet_size = 0;
+        pbft_blocks_rlps.clear();
+        dag_blocks_size = 0;
+        dag_blocks_rlps.clear();
+      }
+
+      dag_blocks_rlps.push_back(dag_rlp.out());
+      dag_blocks_size += dag_rlp.out().size();
     }
+    pbft_block_rlp.appendList(2);  // item #1 - list of dag blocks, item #2 pbft block rlp
+    pbft_block_rlp.appendList(dag_blocks_rlps.size());
+
+    for (auto &dag_block : dag_blocks_rlps) pbft_block_rlp.appendRaw(dag_block);
+
+    pbft_block_rlp.appendRaw(pbft_blocks[pbft_block_idx].rlp());
```

Comment:
are you sure this is ok ? Seems to me that this pbft block is added to the end of dga blocks array. 
 
Isn't this the correct sequence  ? :

  pbft_block_rlp.appendList(2);  // item #1 - list of dag blocks, item #2 pbft block rlp
  pbft_block_rlp.appendRaw(pbft_blocks[pbft_block_idx].rlp());
  
  pbft_block_rlp.appendList(dag_blocks_rlps.size());
  for (auto &dag_block : dag_blocks_rlps) pbft_block_rlp.appendRaw(dag_block);